### DISCUSSION TITLE
emgucv v 4.6.0 use of Mat object

### DISCUSSION THREAD

**Discussion by MaykaS** [emgucv/emgucv#815] [Q&A]
We have found the following info:
https://github.com/emgucv/emgucv/discussions/670 saying that the support in Image<> will stop at v5 , we are moving to replace the Image<> with Mat objects.

Up to this point we used Image<> . to bitmap and linking the BitmapSource to a Wpf object and presenting the image that way.,
As we are moving to use Mat objects we came across : https://www.emgu.com/wiki/files/4.6.0/document/html/M_Emgu_CV_BitmapExtension_ToBitmap_1.htm that we can not use Mat.ToBitmap unless we use this extensionץ

is the BitmapExtension extension the recommended way to move between Mat and an WPF object?
if not, what is the recommended way?
if yes, is this extension planned to be supported with future EMGU versions?
Could you point us to a set of code examples relevant to the EMGU 4.6? We're finding many examples, but most are out of date and irrelevant.
We're interested in image manipulation examples, saving to file, transitioning betweenn different formats, working with image codecs etc

**Comment by profdeveloper231** [comment]
The to bitmap conversion method is useless when you have a save method that allows you to save the image into a file , then you can load the image from the file into a bitmap like below


```
//you have a sample image path 
var path ="imageA.png";
//load the image into emgu cv struct
var image = new Image<Bgr, byte>(path);
//convert image to gray scale
var gray = image.ConvertTo<Gray,byte>();
//you do not need to get the Mat instance of the object above
//save the gray scale image to a temp file
gray.Save("gray.png");
//display the new image in your picture box
pictureBox1.ImageLocation = "gray.png";
```

**Reply by emgucv** [reply]
For conversion from Mat to BitmapSource (WPF), use the Emgu.CV.Wpf nuget package:
https://www.nuget.org/packages/Emgu.CV.Wpf#readme-body-tab

It will be the offcial supported way to convert Mat into BitmapSource moving forward.

```
Mat mat = ... // Create a Mat, load pixel data into it
BitmapSource bmp = mat.ToBitmapSource(); // convert Mat to BitmapSource
```
